libr/core/cmd_print.c

```diff
@@ -3333,6 +3333,59 @@ static bool cmd_print_blocks(RCore *core, const char *input) {
 	return result;
 }
 
+static void list_avaible_plugins(char *path) {
+	RList *files;
+	RListIter *iter;
+	const char *fn;
+	if (path) {
+		files = r_sys_dir (path);
+		r_list_foreach (files, iter, fn) {
+			if (*fn && *fn != '.') {
+				r_cons_println (fn);
+			}
+		}
+		r_list_free (files);
+		free (path);
+	}
+}
+
+static void load_charset_plugin(RCore *core, char *charset_plugin_path) {
+	if (!sdb_query_file (core->anal->charset_db, charset_plugin_path)) {
+		eprintf ("Path: %s not found\n", charset_plugin_path);
+	}
+
+	SdbListIter *iter;
+	SdbKv *kv;
+	SdbList *sdbls = sdb_foreach_list (core->anal->charset_db, true);
+	size_t i = 0;
+
+	core->custom = r_charset_new ();
+	ls_foreach (sdbls, iter, kv) {
+		ut8 *buff_hex = (ut8 *) sdbkv_key (kv);
+		ut8 *buff_char = (ut8 *) sdbkv_value (kv);
+
+		ut64 mylonglong = r_num_get (NULL, (const char *)buff_hex);
+		ut8 *hex = (ut8 *) ut64_new (mylonglong);
+
+		core->custom->custom_charset = add_rune (core->custom->custom_charset, buff_char, hex);
+		i++;
+	}
+	ls_free (sdbls);
+}
+
+static bool cmd_pse(RCore *core, const char *input) {
+	char *path = r_str_r2_prefix (R_JOIN_2_PATHS (R2_SDB, "util"));
+	char *plugin_folder = r_str_newf (R_JOIN_2_PATHS ("%s", "d"), path);
+	const char *plugin_file = r_str_trim_head_ro (input + 3);
```

Comment:
i think pse should take the encoding plugin name from `cfg.encoding` or something like this. 

pse should take one numeric argument, like all the other p subcommands. the data must be taken from the current offset.

Reply:
^-- this is important. also add tests for this

Reply:
maybe we can make rbin provide this cfg.encoding value? cc @condret ?